### PR TITLE
rename BotStateSet -> AutoSaveSetMiddleware

### PR DESCRIPTION
**Pull request by tomlm**

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder/AutoSaveStateMiddleware.cs [hunk 1]
```diff
@@ -12,13 +12,13 @@ namespace Microsoft.Bot.Builder
...
     {
         /// <summary>
-        /// Initializes a new instance of the <see cref="BotStateSet"/> class.
+        /// Initializes a new instance of the <see cref="AutoSaveStateMiddleware"/> class.
         /// </summary>
         /// <param name="botStates">initial list of <see cref="BotState"/> objects to manage.</param>
-        public BotStateSet(params BotState[] botStates)
+        public AutoSaveStateMiddleware(params BotState[] botStates)
         {
             BotStates.AddRange(botStates);
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder/AutoSaveStateMiddleware.cs]: Please add a null check to this. The BotStates is just a list, and will happily accept nulls. We should throw if nothing is passed.

File: libraries/Microsoft.Bot.Builder/AutoSaveStateMiddleware.cs [hunk 2]
```diff
@@ -34,7 +34,7 @@ public BotStateSet(params BotState[] botStates)
         /// </summary>
         /// <param name="botState">botState to manage.</param>
         /// <returns>botstateset for chaining more .Use().</returns>
-        public BotStateSet Use(BotState botState)
+        public AutoSaveStateMiddleware Use(BotState botState)
         {
             BotStates.Add(botState);
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder/AutoSaveStateMiddleware.cs]: Parameter / Null  check.

File: libraries/Microsoft.Bot.Builder/AutoSaveStateMiddleware.cs [hunk 3]
```diff
@@ -12,13 +12,13 @@ namespace Microsoft.Bot.Builder
     ///  Middleware that will call `read()` and `write()` in parallel on multiple `BotState`
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder/AutoSaveStateMiddleware.cs]: This comment, and the one on "OnTurnAsync" are misleading. While this method does have an LoadAsync method, that method needs to be manually called. Given that this is now "AutoSaveSet" we should remove that functionality, or document it more properly.

**Comment by drub0y** [comment]
I love this, the only thing I wanted to point out is that there is still a `LoadAsync` method exposed on this class and it's not called by anything nor do I think it belongs here anymore once you make this change to make it purely about saving state. In fact, I would argue that the `SaveChangesAsync` method not even be exposed publicly any more as it's just an internal detail of the middleware and not an API that people should be calling themselves.

**Pull request merged**